### PR TITLE
 	New automated test to detect when piwik.js minified file is out of sync with original piwik.js

### LINKED ISSUE #8195: New automated test to detect when piwik.js minified file is out of sync with original piwik.js
Issue: it's happened a few times that we modify `js/piwik.js` (the original piwik.js) and that we forget to commit to git the minified version `piwik.js` of this file. 

Usually we detect these problems before the stable release, but it is likely that in the future we will forget and release a new Piwik version with some "missing bug fix" in the Javascript client.

Goal: create an automated test that will fail when the minified of js/piwik.js is different from the piwik.js found in git.

### PR DESCRIPTION
**Pull request by mattab**
fixes  #8195

edit: tests fail on travis (works locally), not sure yet why

### CODE REVIEW

**Comment by mattab** [comment]
FYI Build is now green

**Comment by mattab** [comment]
I'm merging now to get the up to date minified piwik.js - if you have feedback post merge i'll be happy to apply of course!

File: tests/PHPUnit/Integration/ReleaseCheckListTest.php [hunk 1]
```diff
@@ -329,6 +330,18 @@ public function testPiwikJavaScript()
...
+    public function test_piwikJs_minified_isUpToDate()
+    {
+        Http::fetchRemoteFile('https://github.com/downloads/yui/yuicompressor/yuicompressor-2.4.7.zip', PIWIK_DOCUMENT_ROOT .'/tmp/yuicompressor.zip');
+        shell_exec('unzip -n '. PIWIK_DOCUMENT_ROOT .'/tmp/yuicompressor.zip');
+        shell_exec("sed '/<DEBUG>/,/<\/DEBUG>/d' < ". PIWIK_DOCUMENT_ROOT ."/js/piwik.js | sed 's/eval/replacedEvilString/' | java -jar yuicompressor-2.4.7/build/yuicompressor-2.4.7.jar --type js --line-break 1000 | sed 's/replacedEvilString/eval/' | sed 's/^[/][*]/\/*!/' > " . PIWIK_DOCUMENT_ROOT ."/piwik-minified.js");
+        $piwikJsMinified = file_get_contents(PIWIK_DOCUMENT_ROOT ."/piwik-minified.js");
+
+        $piwikJsMinifiedInGit = file_get_contents(PIWIK_DOCUMENT_ROOT . '/piwik.js');
+
+        $this->assertEquals($piwikJsMinified, $piwikJsMinifiedInGit, 'piwik.js minified is out of date - please re-generate the piwik.js. See instructions in js/README');
```
Comment (sgiehl) [inline tests/PHPUnit/Integration/ReleaseCheckListTest.php]: Why aren't you using `assertFileEquals`? that would do the `file_get_contents` automatically.

Comment (mattab) [inline tests/PHPUnit/Integration/ReleaseCheckListTest.php]: good trick! done in https://github.com/piwik/piwik/commit/c881db234a884f127e18dcafac84c9e6407fec17

**Pull request merged**